### PR TITLE
DYN-4765 onboarding step 3 place node

### PR DESCRIPTION
**Pull request by filipeotero**
### Purpose

This PR contains step 3 of the onboarding guide which consists in creating a node.
For this, the user will create a node when the item ByCenterPointRadius is clicked and move to the next step.
If the user gets back to the previous step or moves forward and gets back to this one, the node will be deleted to perform the creation again.

step3PlaceNode

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@QilongTang @Amoursol 

### FYIs

@RobertGlobant20

### CODE REVIEW

File: src/DynamoCoreWpf/UI/GuidedTour/dynamo_guides.json [hunk 1]
```diff
@@ -965,6 +964,110 @@
...
+						"UIElementGridContainer": "sidebarGrid",
+						"WindowElementNameString": "ByCenterPointRadius",
+						"WindowName": "LibraryView"
+					},
+					"CutOffRectArea": {
+						"WidthBoxDelta": 0,
+						"HeightBoxDelta": 0
+					}
+				},
+				"UIAutomation": [
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/GuidedTour/dynamo_guides.json:998]: Can you remind me what this block does?

File: src/DynamoCoreWpf/UI/GuidedTour/dynamo_guides.json [hunk 2]
```diff
@@ -965,6 +964,110 @@
...
+			{
+				"Sequence": 4,
+				"Name": "connectNodes",
+				"Type": "tooltip",
+				"ShowLibrary": false,
+				"Width": 480,
+				"Height": 320,
+				"StepContent": {
+					"Title": "OnboardingGuideWelcomeTitle",
+					"FormattedText": "GetStartedGuideWelcomeText"
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/GuidedTour/dynamo_guides.json:1064]: I am touching this in my PR so not sure if this should be updated as well

File: src/DynamoCoreWpf/UI/GuidedTour/dynamo_guides.json (hunk 1 above)
Comment (filipeotero) [reply src/DynamoCoreWpf/UI/GuidedTour/dynamo_guides.json:998]: Hi @QilongTang, this block is about automating actions in this step. For example, disabling the next button, uncollapse a menu item, subscribing to a specific button in the workspace...

File: src/DynamoCoreWpf/UI/GuidedTour/dynamo_guides.json (hunk 2 above)
Comment (filipeotero) [reply src/DynamoCoreWpf/UI/GuidedTour/dynamo_guides.json:1064]: Oh, I've added this step just for mocking. Do you want me to remove it?

**Comment by filipeotero** [comment]
As commented in the slack channel, we will have some updates in this PR, which include:
* Change the created node from _ByCenterPointRadius_ to _ByOrigin_
* Place the node in a specific position

FYI: @RobertGlobant20 @QilongTang

**Comment by QilongTang** [comment]
> As commented in the slack channel, we will have some updates in this PR, which include:
> 
> * Change the created node from _ByCenterPointRadius_ to _ByOrigin_
> * Place the node in a specific position
> 
> FYI: @RobertGlobant20 @QilongTang

Thanks, I have also merged my changes so you can update your PR now

File: src/DynamoCoreWpf/UI/GuidedTour/StepUIAutomation.cs [hunk 3]
```diff
@@ -92,6 +93,11 @@ public enum UIControlType { MENUITEM, BUTTON, FUNCTION, JSFUNCTION };
         /// This string contains the element to execute the Automatic Function
         /// </summary>
         public string ElementName { get; set; }
+
+        /// <summary>
+        /// This property is the position of the node that will be placed
+        /// </summary>
+        public Point2D NodePosition { get; set; }
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/GuidedTour/StepUIAutomation.cs:100]: For new properties, let's make them internal if possible

File: src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs [hunk 4]
```diff
@@ -630,5 +633,52 @@ internal static void HighlightPort(Step stepInfo, StepUIAutomation uiAutomationD
...
+            //If any backward action is triggered, the created needs to be deleted 
+            if (currentFlow == GuideFlow.BACKWARD && lastCreatedNode != null)
+            {
+                var stepMainWindow = CurrentExecutingStep.MainWindow as Window;
+                stepInfo.DynamoViewModelStep.CurrentSpaceViewModel.Model.RemoveAndDisposeNode(lastCreatedNode);
+            }
+
+            if (enableFunction)
+            {              
+                stepInfo.DynamoViewModelStep.CurrentSpaceViewModel.Model.NodeAdded += func;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs:664]: Would you help me understand how this works? Where is `enableFunction` set?

**Review by QilongTang — APPROVED** [review]
LGTM

**Pull request merged**